orchagent/mirrororch.cpp

```diff
@@ -75,10 +76,145 @@ MirrorOrch::MirrorOrch(TableConnector stateDbConnector, TableConnector confDbCon
     m_fdbOrch->attach(this);
 }
 
+bool MirrorOrch::bake()
+{
+    SWSS_LOG_ENTER();
+
+    // Freeze the route update during orchagent restoration
+    m_freezeUpdate = true;
+
+    deque<KeyOpFieldsValuesTuple> entries;
+    vector<string> keys;
+    m_mirrorTable.getKeys(keys);
+    for (const auto &key : keys)
+    {
+        vector<FieldValueTuple> tuples;
+        m_mirrorTable.get(key, tuples);
+
+        bool active = false;
+        string monitor_port;
+        string vlan_id;
+
+        for (const auto &tuple : tuples)
+        {
+            if (fvField(tuple) == MIRROR_SESSION_STATUS)
+            {
+                active = fvValue(tuple) == MIRROR_SESSION_STATUS_ACTIVE;
+            }
+            else if (fvField(tuple) == MIRROR_SESSION_MONITOR_PORT)
+            {
+                monitor_port = fvValue(tuple);
+            }
+            else if (fvField(tuple) == MIRROR_SESSION_VLAN_ID)
+            {
+                vlan_id = fvValue(tuple);
+            }
+        }
+
+        if (!active)
+        {
+            continue;
+        }
+
+        SWSS_LOG_NOTICE("Found mirror session %s active before warm reboot");
+
+        // Recover saved active session's monitor port
+        m_recoverySessionPortMap.emplace(key, vlan_id + ":" + monitor_port);
+    }
+
+    return Orch::bake();
+}
+
+bool MirrorOrch::postBake()
```

Comment:
this algorithm has a flaw. for example, let's say previous orchagent has a bug which cacluates a wrong session results (wrong port), in this algorithm, the warm reboot will give the same wrong port, and if the route is not update later, the wrong port will not be corrected.

the correct algorithm will need to check if the recorded results is in the newly calculated results or not, if it is in then use it, if it is not, the warm reboot need to correct that error.

Reply:
I found current approach is difficult to achieve above results. 

the best approach is the move mirror orch after route orch in the dotask loop.

Then in the mirror orch dotask loop, caculate the new mirror session results, check if the old results is contained in the new results, if yes, then use old results. It should also results simpler code since the postBake() introduce new logics which can be avoided.

Reply:
It will leads back to the original issue that we are facing - if there are more than 1 mirror sessions require destination update, comparison logic will not be able to handle such cases, and the processes will crash. If it is expected to fix the wrong session results during the warm reboot, such logic shall be introduced after the comparison logic.